pygfx/renderers/wgpu/meshshader_phong.py

```diff
@@ -0,0 +1,287 @@
+        {{ blending_code }}
+        """
+
+        return (
+            self._code_is_orthographic()
+            + self._code_lighting()
+            + self._code_clipping_planes()
+            + self._code_picking()
+            + self._code_misc()
+            + blending_code
+        )
+
+    def code_fragment(self):
+        return """
+        @stage(fragment)
+        fn fs_main(varyings: Varyings, @builtin(front_facing) is_front: bool) -> FragmentOutput {
+            $$ if color_mode == 'vertex'
+                var color_value = varyings.color;
+            $$ elif color_mode == 'normal'
+                var color_value = vec4<f32>((normalize(varyings.normal.xyz) * 0.5 + 0.5), 1.0);
+            $$ else
+                var color_value = u_material.color;
+            $$ endif
+            $$ if use_color_map is defined
+                color_value = color_value * textureSample( t_color_map, s_color_map, varyings.texcoord );
+            $$ endif
+            let albeido = color_value.rgb;
+
+            var material: BlinnPhongMaterial;
+            material.diffuse_color = albeido;
+            material.specular_color = u_material.specular_color.rgb;
+            material.specular_shininess = u_material.shininess;
+            material.specular_strength = 1.0;  //TODO: Use specular_map if exists
+            var reflected_light: ReflectedLight = ReflectedLight(vec3<f32>(0.0), vec3<f32>(0.0), vec3<f32>(0.0), vec3<f32>(0.0));
+            let view_dir = select(
+                normalize(u_stdinfo.cam_transform_inv[3].xyz - varyings.world_pos),
+                ( u_stdinfo.cam_transform_inv * vec4<f32>(0.0, 0.0, 1.0, 0.0) ).xyz,
+                is_orthographic()
+            );
+            var normal = varyings.normal;
+            if (u_material.flat_shading != 0 ) {
+                let u = dpdx(varyings.world_pos);
+                let v = dpdy(varyings.world_pos);
+                normal = normalize(cross(u, v));
+                normal = select(normal, -normal, (select(0, 1, is_front) + u_stdinfo.flipped_winding) == 1);  //?
+            }
+            normal = select(-normal, normal, is_front);  // do we really need this?
+            var geometry: GeometricContext;
+            geometry.position = varyings.world_pos;
+            geometry.normal = normal;
+            geometry.view_dir = view_dir;
+            var i = 0;
+            $$ if num_point_lights > 0
+                loop {
+                    if (i >= {{ num_point_lights }}) { break; }
+                    let point_light = u_point_lights[i];
+                    var light = getPointLightInfo(point_light, geometry);
+                    if (! light.visible) { continue; }
+                    $$ if receive_shadow
+                    if (point_light.cast_shadow != 0){
+                        let shadow = get_cube_shadow(u_shadow_map_point_light, u_shadow_sampler, i, point_light.light_view_proj_matrix, geometry.position, light.direction, point_light.shadow_bias);
```

Comment:
I get a shader error here if I have more than one spot lights in the scene (that also cast shadows). I briefly tried to track the cause, but was not able to find it yet. Naga seems to indicate its the first argument. Is something somehow different between `n=1`  and `n>1` point lights? BTW, multiple directional lights and spot lights does work.

Reply:
Ah, I seem to have forgotten to test "multiple point light shadows". I will investigate this.

Reply:
Bug have fixed.

Reply:
Confirmed 👍